packages/auth-nextjs/src/app/index.ts

```diff
@@ -0,0 +1,551 @@
+                  error: null,
+                  tokenData: null,
+                  isSignUp: true,
+                });
+              }
+              return onBuiltinUICallback({
+                error: new Error("no pkce code in response"),
+              });
+            }
+            const verifier = req.cookies.get(
+              this.options.pkceVerifierCookieName
+            )?.value;
+            if (!verifier) {
+              return onBuiltinUICallback({
+                error: new Error("no pkce verifier cookie found"),
+              });
+            }
+            const isSignUp =
+              req.nextUrl.searchParams.get("isSignUp") === "true";
+            let tokenData: TokenData;
+            try {
+              tokenData = await (await this.core).getToken(code, verifier);
+            } catch (err) {
+              return onBuiltinUICallback({
+                error: err instanceof Error ? err : new Error(String(err)),
+              });
+            }
+            cookies().set({
+              name: this.options.authCookieName,
+              value: tokenData.auth_token,
+              httpOnly: true,
+              sameSite: "lax",
+            });
+            cookies().delete(this.options.pkceVerifierCookieName);
+
+            return onBuiltinUICallback({ error: null, tokenData, isSignUp });
+          }
+          case "builtin/signin":
+          case "builtin/signup": {
+            const pkceSession = (await this.core).createPKCESession();
+            cookies().set({
+              name: this.options.pkceVerifierCookieName,
+              value: pkceSession.verifier,
+              httpOnly: true,
+            });
+            return redirect(
+              params.auth[params.auth.length - 1] === "signup"
+                ? pkceSession.getHostedUISignupUrl()
+                : pkceSession.getHostedUISigninUrl()
+            );
+          }
+          case "signout": {
+            if (!onSignout) {
+              throw new Error(`'onSignout' auth route handler not configured`);
+            }
+            cookies().delete(this.options.authCookieName);
+            return onSignout();
+          }
+          default:
+            return new Response("Unknown auth route", {
+              status: 404,
+            });
+        }
+      },
+      POST: async (
+        req: NextRequest,
+        { params }: { params: { auth: string[] } }
+      ) => {
+        switch (params.auth.join("/")) {
+          case "emailpassword/signin": {
+            if (!onEmailPasswordSignIn) {
+              throw new Error(
+                `'onEmailPasswordSignIn' auth route handler not configured`
+              );
+            }
+            let tokenData: TokenData;
+            try {
+              const [email, password] = _extractParams(
+                await _getReqBody(req),
+                ["email", "password"],
+                "email or password missing from request body"
+              );
```

Comment:
We also need a `verify_url` here, otherwise it falls back to the built-in UI verify.

Reply:
I can't see that we use `verify_url` anywhere in the `authenticate` endpoint? (https://github.com/edgedb/edgedb/blob/master/edb/server/protocol/auth_ext/http.py#L473)

Reply:
Oh, sorry, I meant to put this on the `/signup` endpoint: https://github.com/edgedb/edgedb/blob/57b82842d02e2c946754baa22248b2d381b85b0e/edb/server/protocol/auth_ext/http.py#L429-L431

Reply:
I see now that it's the third argument to the `signupWithEmailPassword` 👍